controllers/workspace/devworkspace_controller.go

```diff
@@ -240,7 +242,27 @@ func (r *DevWorkspaceReconciler) Reconcile(req ctrl.Request) (reconcileResult ct
 		return reconcile.Result{Requeue: true}, nil
 	}
 
+	annotate.AddURLAttributesToEndpoints(&workspace.Spec.Template, routingStatus.ExposedEndpoints)
+
+	// Step three: provision a configmap on the cluster to mount the flattened devfile in deployment containers
+	err = metadata.ProvisionWorkspaceMetadata(devfilePodAdditions, clusterWorkspace, workspace, &clusterAPI)
+	if err != nil {
+		switch provisionErr := err.(type) {
+		case *metadata.NotReadyError:
+			reqLogger.Info(provisionErr.Message)
+			return reconcile.Result{Requeue: true, RequeueAfter: provisionErr.RequeueAfter}, nil
+		case *metadata.ProvisioningError:
+			reqLogger.Info(fmt.Sprintf("DevWorkspace start failed: %s", provisionErr))
+			reconcileStatus.Phase = devworkspace.WorkspaceStatusFailed
+			reconcileStatus.Conditions[devworkspace.WorkspaceFailedStart] = fmt.Sprintf("Error provisioning storage: %s", provisionErr)
```

Comment:
it's not storage, right? )

Reply:
Yup -- copy paste :)

Fixed in d1b4a9a1